lib/mruby/build/command.rb

```diff
@@ -311,11 +311,11 @@ def run_reset_hard(dir, checksum_hash)
     end
 
     def commit_hash(dir)
-      `#{@command} --git-dir #{shellquote(dir +'/.git')} --work-tree #{shellquote(dir)} rev-parse --verify HEAD`.strip
+      `#{@command} --git-dir \"#{shellquote(dir +'/.git')}\" --work-tree \"#{shellquote(dir)}\" rev-parse --verify HEAD`.strip
```

Comment:
I think it is preferable to modify the `#shellquote` method to handle double-quoting.
The same applies to the `#current_branch` method below.